packages/web-app-files/src/store/actions.js

```diff
@@ -144,11 +149,47 @@ export default {
       value: computeShareTypes(state.currentFileOutgoingShares)
     })
   },
-  loadCurrentFileOutgoingShares(context, { client, path }) {
+  loadCurrentFileOutgoingShares(context, { client, path, space }) {
     context.commit('CURRENT_FILE_OUTGOING_SHARES_SET', [])
     context.commit('CURRENT_FILE_OUTGOING_SHARES_ERROR', null)
     context.commit('CURRENT_FILE_OUTGOING_SHARES_LOADING', true)
 
+    if (space) {
+      const promises = []
+      const spaceShares = []
+
+      for (const permission of space.spacePermissions) {
+        for (const {
+          user: { id }
+        } of permission.grantedTo) {
+          promises.push(
+            client.users.getUser(id).then((resolved) => {
+              spaceShares.push(
+                buildSpaceShare(
+                  {
+                    ...resolved.data,
+                    role: permission.roles[0]
+                  },
+                  space.id
+                )
+              )
+            })
+          )
+        }
+      }
+
+      return Promise.all(promises)
```

Comment:
Just some food for thought: for now this should be fine because we're loading either space members or shares, but never both at the same time. However, this will change in the near future and then we want to display space members and regular shares side by side. Merging the responses will be needed eventually. For now this is fine.